rclcpp/include/rclcpp/context.hpp

```diff
@@ -261,6 +340,16 @@ class Context : public std::enable_shared_from_this<Context>
 
   std::vector<OnShutdownCallback> on_shutdown_callbacks_;
   std::mutex on_shutdown_callbacks_mutex_;
+
+  /// Condition variable for timed sleep (see sleep_for).
+  std::condition_variable interrupt_condition_variable_;
+  /// Mutex for protecting the global condition variable.
+  std::mutex interrupt_mutex_;
+
+  /// Mutex to protect sigint_guard_cond_handles_.
+  std::mutex interrupt_guard_cond_handles_mutex_;
+  /// Guard conditions for interrupting of associated wait sets on interrupt_all_wait_sets().
+  std::map<rcl_wait_set_t *, rcl_guard_condition_t> interrupt_guard_cond_handles_;
```

Comment:
just my 2cts, but a `unordered_map` has generally a faster lookup and does not need to reshuffle the tree when `insert` or `erase` is called on it.